### DISCUSSION TITLE
render_in + Bootstrap Modal -> cannot selected an option

### DISCUSSION THREAD

**Discussion by sandrods** [josefarias/hotwire_combobox#84] [Q&A]
I'm trying to use a combobox inside a bootstrap modal. 
It works correctly the using  default rendering.
When specifying `{render_in: partial: ...}`, the list is correctly loaded with the rendered options (with the partial), but on selecting an item, the combobox remains empty (don't display the selection and the value remains nil).
Putting the same combobox outside the modal and everything works.

Am I missing something here?

**Comment by josefarias** [comment]
Hey @sandrods 

Can you share the part of your template where you're creating the combobox? I want to see how it's set up. Especially whether it's async.

Also, if you could provide a minimal reproduction that'd be super useful. Could be a new rails app with bootstrap set up. Nothing fancy. Just want some broken example to tinker with and fix.

EDIT: also, what version of the gem are you using?

Thanks!

**Comment by sandrods** [comment]
Hello @josefarias. I'll try to create a simple isolated app later, but for now I can tell you this

-> I'm using the latest version (0.1.40)

-> The combobox is remote, yes
```ruby 
   # view code
   <%= f.combobox :destinatario, combobox_usuarios_path, class: "form-control" %>
```
```ruby 
   # controller code
    render turbo_stream: helpers.async_combobox_options(
      @usuarios,
      next_page: @pagy.next,
      render_in: {partial: "combobox/usuario", formats: [:html]}
    )
```

**Reply by sandrods** [reply]
Also, there's another side effect of using render_in that I hadn't noticed before: the pagination has stopped working. It no longer fetches the next page upon reaching the bottom of the list. Removing the render_in makes it work again.
I know it's piling up a bunch of problems. I'll create a sample app as soon as possible

**Comment by josefarias** [comment]
@sandrods yeah these seem like bugs. Went ahead and filed two issues to keep track of them as such. Closing this for now so we can track them separately:

https://github.com/josefarias/hotwire_combobox/issues/91
https://github.com/josefarias/hotwire_combobox/issues/90

**Reply by josefarias** [reply]
Hey @sandrods, regarding pagination, I was able to set up a bootstrap app and load async options with render_in that are paginated using the pagy gem. Seems to work fine for me! Is it possible you only have a single results page? Closed the issue for now, but feel free to comment there with reproduction steps. I'm writing you here because GH won't let me tag you in the issue.

Regarding the failure to select from within a modal, it seems like bootstrap [hijacks click events](https://github.com/twbs/bootstrap-rubygem/blob/90467dee09b0c6e52e7d58de867148fda0dd9f49/assets/javascripts/bootstrap/modal.js#L284-L308) that happen inside the modal and then does some delegation logic. I think this is somehow shifting the focus from the combobox to the modal, and thus calling `closeOnFocusOutside` before the option is selected. This seems like an interaction that's brought about by bootstrap's implementation so I'm not inclined to dive super deep into it right now.

Unfortunately, I'll close the modal issue as "wontfix" until this starts affecting enough HotwireCombobox users or more information comes to light as to what exactly in bootstrap's modal is causing problems. I know that's probably not what you wanted to hear. I'm sorry! I'm just one person maintaining this library so I can't prioritize digging into bootstrap's code right now.

**Reply by sandrods** [reply]
No worries, @josefarias . I certainly can relate to being a one person maintainer. It has its ups and downs. 

Regarding the pagination, I'll double check the code as I'm sure there's multiple pages. Will (finally!) create a test app for confirming (or not) the issue and will send you in case I can reproduce it.

As for the Bootstrap modal, that's a real bummmer. tom-select and selectize had this issues with Bootstrap also. I think even good ol'select2 had them. I'll try to look into it and see if I can help in any way.

Anyway, thanks for that awesome work on solving that last usolved UI challenge for us rails FE devs. Keep up the good work and let me know if there's any way I can help

**Reply by josefarias** [reply]
Thanks for understanding! If those libraries have fixed those issues then there might be some prior art that will help us fix, for sure.

I'll take a peek at those libraries' histories but would certainly appreciate if you could look into it, too.

Thank you!

**Reply by aroth** [reply]
Hi again @josefarias. This issue is also biting us using tailwindcss-stimulus-components `Modal`. https://github.com/excid3/tailwindcss-stimulus-components 

Just a heads up for when/if you're able to take a closer look.

**Reply by josefarias** [reply]
Hey @aroth. This seemed to be specific to bootstrap. Can you share a reproduction app so I can take a closer look?

**Reply by josefarias** [reply]
@aroth just going through my notifications here — making sure you got this before I clear this one.

I'd love to take a look if you have a repro. Thanks!

**Reply by aroth** [reply]
@josefarias Good to clear. I will end up getting back to this with a repro in the next couple months I suspect, when I double back on this bit of work. I found a workaround and other higher priority tasks came up. Again, thanks for your great work on this.

**Reply by shayani** [reply]
HI @josefarias. Thanks for this awesome gem. I'm passing by just to say that also I'm having the same issue with Boostrap modals. Even a simple combobox (`combobox_tag("user", User.all)`) doesn't allow me to select an option. 

You said that will keep this issue as a 'wont fix' until have enough users, so I'm adding myself to this list of users ;) 

Thanks!
